tools/eslint-rules/new-with-error.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @fileoverview Require `throw new Error()` rather than `throw Error()`
+ * @author Rich Trott
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+  return {
+    'ThrowStatement': function(node) {
+      const arg = node.argument;
+      if ((arg.type === 'CallExpression') && (arg.callee.name === 'Error')) {
```

Comment:
@cjihrig I'll make that adjustment.

@Fishrock123 You're after this: `/^([A-Z]\w*)?Error$/` That assumes `FooError()` will never be the name of a function that returns an Error object and will only be a constructor for an Error object.